solutions/devsprint-bruno-faganello-1/FinanceApp/Modules/ActivityDetails/ActivityDetailsRouter.swift

```diff
@@ -0,0 +1,33 @@
+//
+//  ActivityDetailsRouter.swift
+//  FinanceApp
+//
+//  Created by Victor Catão on 14/03/22.
+//
+
+import UIKit
+
+// MARK: - ActivityDetailsRouter
+
+final class ActivityDetailsRouter: ActivityDetailsRouterProtocol {
+    
+    // MARK: Public Methods
+
+    static func createModule() -> UIViewController {
+        let viewController = ActivityDetailsViewController()
+
+        let presenter: ActivityDetailsPresenterProtocol & ActivityDetailsInteractorDelegate = ActivityDetailsPresenter()
```

Comment:
Eu segui o Sample aqui. Por que foi usado assim? `ActivityDetailsPresenterProtocol & ActivityDetailsInteractorDelegate`

Achei estranho porque o ViewController.presenter é só um `ActivityDetailsPresenterProtocol`, não exige que seja `ActivityDetailsInteractorDelegate` também. Se for esse mesmo o caso, teria uma forma de especificar melhor o que essa variável presenter precisa ser? Talvez criar um typealias? Ou ela não precisaria ser desses 2 tipos?